overlord/snapstate/handlers.go

```diff
@@ -1166,10 +1240,13 @@ func (m *SnapManager) doCopySnapData(t *state.Task, _ *tomb.Tomb) (err error) {
 		return copyDataErr
 	}
 
-	action := triggeredMigration(newInfo.Base, opts)
-	var undo func()
+	var oldBase string
+	if oldInfo != nil {
+		oldBase = oldInfo.Base
+	}
 
-	switch action {
+	var undo func()
+	switch triggeredMigration(oldBase, newInfo.Base, opts) {
```

Comment:
we should decide whether to error or at least log/warn if we get a core22 -> old downgrade here?

Reply:
ok, it should be a log on the task I suppose